app/controllers/posts_controller.rb

```diff
@@ -9,4 +9,50 @@ def show
     @post = Post.find(params[:id_post])
     @comments = Comment.where(post_id: @post.id)
   end
+
+  def new
+    @post = Post.new
+    @user = current_user
+    respond_to do |format|
+      format.html { render :new, locals: { post: @post } }
+    end
+  end
+
+  def create
+    @post = Post.new(params.require(:post).permit(:title, :text))
+    @post.author = current_user
```

Comment:
- [ **OPTIONAL** ] For best practices reasons, I think it would be better if you you use `params` method when creating a post. Once you have created the `params` methos method you make it as private. You can refer this link :point_right: for short **[remind about strong parameters lesson](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides). For example:

 ```ruby
  ...
  def create
      @post = @current_user.posts.new(post_params)

      respond_to do |format|
        if @post.save
          format.html { redirect_to @post }
        else
          format.html { render :new }
        end
      end
    end
   ...

 private 

    def post_params
      params.require(:post).permit(:title, :text)
    end

	  ...
```
**Important:** **I think it would be better if you apply this approach in your ``controllers/comments_controller.rb`` file when creating a comment.**